warp10/src/main/java/io/warp10/script/aggregator/Percentile.java

```diff
@@ -138,65 +150,99 @@ public int compare(Integer idx1, Integer idx2) {
     }
 
     int nonNullLength = values.length - nullCounter;
+
     //
     // Compute rank
     //
 
-    int n = (int) Math.round(0.5 + this.percentile * nonNullLength / 100.0) - 1;
+    double p = this.percentile / 100.0D;
+
+    double pn = p * nonNullLength;
 
     if (!this.interpolate) {
-      if (n >= nonNullLength) {
-        n--;
+      // Type 1 in Hyndman, Rob & Fan, Yanan. (1996). Sample Quantiles in Statistical Packages. The American Statistician.
+      int j = (int) Math.floor(pn);
+      if (pn > j) {
+        j++;
       }
```

Comment:
Can't you use just Math.ceil() instead ?

Reply:
Actually I think it's cleaner to do
Math.floor(p*(nonNullLength -1))
so there is no need to reindex from 1-indexed to 0-indexed

Reply:
That would lead to a different result. For instance with `nonNullLength = 4` and `p = 0.9`:
```
Math.ceil(p * nonNullLength) - 1 = 3
Math.floor(p * (nonNullLength - 1)) = 2
```